### PR TITLE
Remove misleading MSSQL information from the docs

### PR DESCRIPTION
**Pull request by potiuk**
Some of the past cherry-picks brought MSSQL information to Airflow
docs in stable version. This should be removed.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Review by mik-laj — APPROVED** [review]

File: docs/apache-airflow/howto/set-up-database.rst [hunk 1]
```diff
@@ -27,14 +27,13 @@ The document below describes the database engine configurations, the necessary c
 Choosing database backend
 -------------------------
 
-If you want to take a real test drive of Airflow, you should consider setting up a database backend to **PostgreSQL**, **MySQL**, or **MSSQL**.
+If you want to take a real test drive of Airflow, you should consider setting up a database backend to **PostgreSQL**, **MySQL**.
```
Comment (malthe) [inline docs/apache-airflow/howto/set-up-database.rst:30]: Or?

Comment (potiuk) [reply docs/apache-airflow/howto/set-up-database.rst:30]: too late :) . But yeah. It makes sense also without :)

**Comment by jedcunningham** [comment]
I certainly don't have a complete history here, but it does look like it was intentionally added before 2.2.0 and wasn't a result of a rogue cherry-pick:

https://github.com/apache/airflow/issues/17018

**Comment by potiuk** [comment]
> I certainly don't have a complete history here, but it does look like it was intentionally added before 2.2.0 and wasn't a result of a rogue cherry-pick:
> 
> #17018

Right. I have not checked exactly (Assumed it was a cherry-pick) but it is misleading our users (as we do not support MsSQl in any released version of Airflow yet)

**Comment by potiuk** [comment]
BTW. @jedcunningham -  I assumed it was MY wrong cherry-pick to be precise :)

**Comment by jedcunningham** [comment]
All good, I just started poking around because I didn't know how it got there either and was worried I goofed up 👍

I guess for 2.3.0 we have to either decide to adopt support, tag it as experimental in those howto docs, or remove the docs from main, otherwise we will have the same issue.

**Comment by potiuk** [comment]
> I guess for 2.3.0 we have to either decide to officially support it, tag it as experimental in those howto docs, or remove the docs from main, otherwise we will have the same issue.

Absolutely. Taking into account a number of issues, experimental I'd say :)

**Pull request merged**